src/components/Contexts/FormDataProvider.jsx

```diff
@@ -189,64 +32,22 @@ const FormDataProvider = ({ children }) => {
   })
 
   const formSubmit = () => {
-    const qualification = Qualification.getByLabel(stepperDialogTitle)
-    const { metadata } = formData
     ;(async () => {
       try {
-        const contacts = formData.contacts
+        const qualification = Qualification.getByLabel(stepperDialogTitle)
         const { _id: appFolderID } = await getOrCreateAppFolderWithReference(
           client,
           t
         )
-        const fileCollection = client.collection(FILES_DOCTYPE)
 
-        let pdfDoc = await PDFDocument.create()
-        const isMultiPage = formData.data.some(
-          ({ fileMetadata }) => fileMetadata.multipage
-        )
-
-        for (const { file, fileMetadata } of formData.data) {
-          await addFileToPdf({
-            pdf: pdfDoc,
-            fileToAdd: file
-          })
-
-          if (!isMultiPage) {
-            await savePdfToCozy({
-              pdf: pdfDoc,
-              pdfMetadata: fileMetadata,
-              formMetadata: metadata,
-              currentDefinition,
-              qualification,
-              contacts,
-              fileCollection,
-              appFolderID,
-              f,
-              t,
-              scannerT,
-              client
-            })
-
-            pdfDoc = await PDFDocument.create()
-          }
-        }
-
-        if (isMultiPage) {
-          await savePdfToCozy({
-            pdf: pdfDoc,
-            pdfMetadata: {},
-            formMetadata: metadata,
-            currentDefinition,
-            qualification,
-            contacts,
-            fileCollection,
-            appFolderID,
-            f,
-            t,
-            scannerT,
-            client
-          })
-        }
+        await createPdfAndSave({
+          formData,
+          qualification,
+          currentDefinition,
+          appFolderID,
+          client,
+          i18n: { t, f, scannerT }
```

Comment:
Do we still really need `scannerT`?

Reply:
Yes, the name may be confusing, it now comes from `model.document` from `cozy-client`
https://github.com/cozy/mespapiers/blob/master/src/components/Contexts/ScannerI18nProvider.jsx#L3-L6